src/netcdf_writer.jl

```diff
@@ -546,30 +601,49 @@ NVTX.@annotate function write_field!(
     # position ever if we are writing the file for the first time)
     time_index = temporal_size + 1
 
-    # TODO: Use ITime here
-    nc["time"][time_index] = float(t)
-    nc["time_bnds"][:, time_index] =
-        time_index == 1 ? [zero(float(t)); float(t)] :
-        [nc["time"][time_index - 1]; float(t)]
-
-    # FIXME: We are hardcoding p.start_date !
-    # FIXME: We are rounding t
-    if !isnothing(start_date)
+    if isnothing(diagnostic.output_writer.buffers)
         # TODO: Use ITime here
-        curr_date = start_date + Dates.Millisecond(round(1000 * float(t)))
-        date_type = typeof(curr_date) # not necessarily a Dates.DateTime
-        nc["date"][time_index] = curr_date
-        nc["date_bnds"][:, time_index] =
-            time_index == 1 ? [start_date; curr_date] :
-            [date_type(nc["date"][time_index - 1]); curr_date]
-    end
+        nc["time"][time_index] = float(t)
+        nc["time_bnds"][:, time_index] =
+            time_index == 1 ? [zero(float(t)); float(t)] :
+            [nc["time"][time_index - 1]; float(t)]
+
+        # FIXME: We are hardcoding p.start_date !
+        # FIXME: We are rounding t
+        if !isnothing(start_date)
+            # TODO: Use ITime here
+            curr_date = start_date + Dates.Millisecond(round(1000 * float(t)))
+            date_type = typeof(curr_date) # not necessarily a Dates.DateTime
+            nc["date"][time_index] = curr_date
+            nc["date_bnds"][:, time_index] =
+                time_index == 1 ? [start_date; curr_date] :
+                [date_type(nc["date"][time_index - 1]); curr_date]
+        end
+
+        colons = ntuple(_ -> Colon(), length(dim_names))
+        v[time_index, colons...] = interpolated_field
 
-    colons = ntuple(_ -> Colon(), length(dim_names))
-    v[time_index, colons...] = interpolated_field
+    else
+        buffer = get(diagnostic.output_writer.buffers, diagnostic, nothing)
+        if isnothing(buffer)
+            diagnostic.output_writer.buffers[diagnostic] = NetCDFBuffer(
+                nc,
+                var.short_name,
+                start_date,
+                t,
+                interpolated_field,
+                max_capacity = writer.buffer_max_capacity,
+            )
```

Comment:
```suggestion
        if isnothing(buffer)
            buffer = NetCDFBuffer(nc, var.short_name, start_date, t, interpolated_field,
                                  max_capacity = writer.buffer_max_capacity)
            diagnostic.output_writer.buffers[diagnostic] = buffer
        else
            push!(buffer, t, interpolated_field)
        end
```
I think the buffer variable needs to be set if `isnothing(buffer)`